test/bootboot_test.rb

```diff
@@ -209,6 +209,54 @@ def test_bootboot_command_initialize_the_next_lock_and_update_the_gemfile
     end
   end
 
+  def test_bundle_install_with_different_ruby_updating_gemfile_next_lock_succeeds
+    write_gemfile do |file, _dir|
+      FileUtils.cp("#{file.path}.lock", gemfile_next(file))
+      File.write(file, <<-EOM, mode: 'a')
+        if ENV['DEPENDENCIES_NEXT']
+          ruby '9.9.9'
+        else
+          ruby '#{RUBY_VERSION}'
+        end
+      EOM
+
+      run_bundler_command('bundle install', file.path)
+
+      assert_equal(
+        RUBY_VERSION,
+        Bundler::Definition.build(
+          file.path, "#{file.path}.lock", false
+        ).ruby_version.gem_version.to_s
+      )
+
+      with_env_next do
+        assert_equal(
+          "9.9.9",
+          Bundler::Definition.build(
+            file.path, gemfile_next(file), false
+          ).ruby_version.gem_version.to_s
+        )
+      end
+    end
+  end
+
+  def test_bundle_install_with_different_ruby_for_installing_gemfile_next_lock_fails
+    write_gemfile do |file, _dir|
+      FileUtils.cp("#{file.path}.lock", gemfile_next(file))
+      File.write(file, <<-EOM, mode: 'a')
+        if ENV['DEPENDENCIES_NEXT']
+          ruby '9.9.9'
+        else
+          ruby '#{RUBY_VERSION}'
+        end
+      EOM
+
+      assert_raises BundleInstallError do
```

Comment:
Can you add an assertion on the error message raised as it wasn't clear at first why this would raise an error